cloud/awsprovider.go

```diff
@@ -137,22 +152,46 @@ func (aws *AWS) KubeAttrConversion(location, instanceType, operatingSystem strin
 	return region + "," + instanceType + "," + operatingSystem
 }
 
+type awsKey struct {
+	SpotLabelName  string
+	SpotLabelValue string
+	Labels         map[string]string
+	ProviderID     string
+}
+
+func (k *awsKey) ID() string {
+	parsedProviderID := strings.Split(k.ProviderID, "/")
+	if len(parsedProviderID) == 5 { // It's of the form aws:///us-east-2a/i-0fea4fd46592d050b and we want i-0fea4fd46592d050b, if it exists
+		return parsedProviderID[4]
+	}
+	return ""
+}
+
 // GetKey maps node labels to information needed to retrieve pricing data
-func (aws *AWS) GetKey(labels map[string]string) string {
-	instanceType := labels["beta.kubernetes.io/instance-type"]
-	operatingSystem := labels["beta.kubernetes.io/os"]
-	region := labels["failure-domain.beta.kubernetes.io/region"]
-	if l, ok := labels["lifecycle"]; ok && l == "EC2Spot" {
+func (k *awsKey) Features() string {
+	instanceType := k.Labels["beta.kubernetes.io/instance-type"]
```

Comment:
Might I point out that there are constants for these: [v1.LabelInstanceType](https://godoc.org/k8s.io/api/core/v1#LabelInstanceType), and one only needs to fall back to the `beta.kubernetes.io/os` if [`v1.LabelOSStable`](https://godoc.org/k8s.io/api/core/v1#LabelOSStable) is missing

Reply:
Thanks!